src/v2/Apps/Home/Components/HomeTroveArtworksRail.tsx

```diff
@@ -0,0 +1,118 @@
+import { ContextModule } from "@artsy/cohesion"
+import {
+  Box,
+  Skeleton,
+  SkeletonBox,
+  SkeletonText,
+  Spacer,
+} from "@artsy/palette"
+import { createFragmentContainer, graphql } from "react-relay"
+import { ShelfArtworkFragmentContainer } from "v2/Components/Artwork/ShelfArtwork"
+import { Rail } from "v2/Components/Rail/Rail"
+import { SystemQueryRenderer } from "v2/System/Relay/SystemQueryRenderer"
+import { extractNodes } from "v2/Utils/extractNodes"
+import { HomeTroveArtworksRail_viewer } from "v2/__generated__/HomeTroveArtworksRail_viewer.graphql"
+import { HomeTroveArtworksRailQuery } from "v2/__generated__/HomeTroveArtworksRailQuery.graphql"
+
+interface HomeTroveArtworksRailProps {
+  viewer: HomeTroveArtworksRail_viewer
+}
+
+export const HomeTroveArtworksRail: React.FC<HomeTroveArtworksRailProps> = ({
+  viewer,
+}) => {
+  const artworks = extractNodes(viewer.saleArtworksConnection)
+  if (artworks.length === 0) {
+    return null
+  }
```

Comment:
this is so #superminor as to ignore, but still a tip 😄 

As you start writing more and more code, a good code writing habit is to add some linebreaks for readability. Like... think of each statement as a paragraph of sorts. You've got your variables declared on top (new paragraph), then you're checking a condition (new paragraph) then you're returning some JSX:

```tsx 
const artworks = extractNodes(viewer.saleArtworksConnection)

if (artworks.length === 0) {
  return null
}

return (
  ... 
)
```

When everything is compressed together it can often feel like a run-on sentence, or an extra long paragraph -- hard to understand the gist of at a glance.

Reply:
Ditto on this advice.